rules/CodingStyle/Application/UseImportsAdder.php

```diff
@@ -253,4 +291,77 @@ private function isCurrentNamespace(string $namespaceName, ObjectType $objectTyp
 
         return ! \str_contains($afterCurrentNamespace, '\\');
     }
+
+    /**
+     * @param Stmt[] $stmts
+     * @param Use_[] $nodesToAdd
+     * @return Stmt[]
+     */
+    private function insertUseNodesInStatements(
+        array $stmts,
+        array $nodesToAdd,
+        int $position = 0,
+        bool $addSpace = true
+    ): array {
+        $importInsertSorted = SimpleParameterProvider::provideBoolParameter(Option::IMPORT_INSERT_SORTED);
+        if ($importInsertSorted && isset($stmts[$position])
+            && ($stmts[$position] instanceof Use_ || $stmts[$position] instanceof GroupUse)) {
+            foreach ($nodesToAdd as $nodeToAdd) {
+                do {
+                    $useListPosition = 0;
+                    $prefix = '';
+                    if (! isset($stmts[$position])
+                        || (! $stmts[$position] instanceof Use_ && ! $stmts[$position] instanceof GroupUse)) {
+                        break;
+                    }
+
+                    if ($stmts[$position] instanceof GroupUse) {
+                        $prefix = $stmts[$position]->prefix->toString() . '\\';
+                    }
+
+                    $parentUseType = $stmts[$position]->type;
+                    $newUseType = $nodeToAdd->type;
+                    foreach ($stmts[$position]->uses as $use) {
+                        $currentUseType = $parentUseType === Use_::TYPE_UNKNOWN ? $use->type : $parentUseType;
+
+                        if ($newUseType < $currentUseType) {
+                            break 2;
+                        }
+
+                        if ($newUseType === $currentUseType
+                            && $prefix . $use->name->toString() > $nodeToAdd->uses[0]->name->toString()) {
+                            break 2;
+                        }
+
+                        ++$useListPosition;
+                    }
+
+                    ++$position;
+                } while (true);
+
+                if ($useListPosition === 0) {
+                    array_splice($stmts, $position, 0, [$nodeToAdd]);
+                } else {
+                    assert($stmts[$position] instanceof Use_ || $stmts[$position] instanceof GroupUse);
+                    if ($prefix !== '') {
+                        $nodeToAdd->uses[0]->name =
+                            new Name(substr($nodeToAdd->uses[0]->name->toString(), strlen($prefix)));
+                        $nodeToAdd->uses[0]->type = $nodeToAdd->type;
+                    }
+
+                    array_splice($stmts[$position]->uses, $useListPosition, 0, [$nodeToAdd->uses[0]]);
+                }
+
+                ++$position;
+            }
+        } else {
+            if ($addSpace) {
```

Comment:
If there were no existing use statements or the option to insert sorted was not set, we just insert the whole block of statements in one go. If the option to sort the statements was set, the statements will already be sorted.

Also, in some cases we may need to add a `Nop` statement after our list of statements